azpubsub/api/src/main/scala/azpubsub/kafka/security/auth/TokenValidator.scala

```diff
@@ -0,0 +1,21 @@
+package azpubsub.kafka.security.auth
+
+import java.util.Map
+
+/**
+  * Interface to validate base64 encoded token from Kafka client.
+  */
+trait TokenValidator {
```

Comment:
Let's use Java interface and avoid Scala code unless necessary as kafka community is also trying to move away from scala where ever possible.

Reply:
Firstly, really appreciate your time in reviewing the code.  
This is code is completely copied from kafka 1.1. I decided to use Scala instead of Java because the component using this interface/trait is Scala.  
If we need to change, we should change in both Kafka 1.1 and Kafka 2.3. 

But let's take one step back. As we agreed in the retrospective meeting, we should KEEP IT STUPID SIMPLE (KISS).  I really don't see it  is necessary to make this change. 
The code (including the code in Kafka 1.1) has been there for almost 6 months, and we used to discuss about Scala/Java. We really don't need to restart another discussion or debate on this topic.
 
And more importantly, given that the code has been there for so long time, we really expect to see some high quality comments instead of very high level comments. We really expect engineers from the team can dive deep into the details to understand how Kafka code is working when using security token, how our code is validate token/authorize resources, and be able to find out potential issues in our new code.